### ISSUE TITLE
"Current user is not logged in" (997) when accessing ocs/v2.php/cloud/user with a bearer token

### ISSUE THREAD

**Issue by Dagefoerde** [nextcloud/server#12665]
Discovered on Friday; reproduced today using Nextcloud 15 RC 1.

### Steps to reproduce
1. Register & configure an OAuth 2 client
2. Obtain an access token
3. Access `ocs/v2.php/cloud/user?format=json` with that access token

### Expected behaviour

JSON representation of the user object; particularly with the username in the `ocs/data/id` part.

### Actual behaviour

```
{
    "ocs": {
        "data": [],
        "meta": {
            "message": "Current user is not logged in",
            "status": "failure",
            "statuscode": 997
        }
    }
}
```

### Server configuration

**Nextcloud version:** Nextcloud 15 RC1.

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh

**Where did you install Nextcloud from:** https://download.nextcloud.com/server/prereleases/nextcloud-15.0.0RC1.zip

**Comment by Dagefoerde** [comment]
I checked: This worked without issue in Nextcloud 15 beta 2, but breaks in 15 RC 1.

cc @rullzer, thx

**Comment by rullzer** [comment]
Mmm it is weird it is broken in RC1 as we fixed something with bearer tokens there.
I'll see if i can trigger it.

**Comment by rullzer** [comment]
@Dagefoerde I can't reproduce this.

I create a token. And then on curl do:

`curl -H 'Authorization: Bearer <CODE> <server>/ocs/v2.php/cloud/user`

This breaks on your system?

**Comment by Dagefoerde** [comment]
@rullzer nevermind. Sorry to bother you with this. There was a Moodle bug report in https://tracker.moodle.org/browse/MDL-64183. To reproduce this, I installed a fresh Nextcloud 15 pre-release because I wanted to test against that anyway. I was able to reproduce it, but only because the new server's Apache config was wrong. Theirs was, too.

Nextcloud 15 RC 1 is fine and 14.0.4 is, too. Sorry (and thanks for checking!)
